src/composables/useSpacefinderSupabase.ts

```diff
@@ -32,5 +34,56 @@ export default function useSpacefinderSupabase() {
     buildingsRealtimeChannel.subscribe();
   }
 
-  return { getBuildingsOccupancyCurrent, subscribeToBuildingsOccupancy };
+  async function getSpacesOccupancyCurrent() {
+    const { data, error, status } = await client
+      .from("spaces_latest_states")
+      .select("*");
+    const activeDevicesPerBuildingAndSpace: Record<
+      number,
+      Record<string, number>
+    > = {};
+    if (data) {
+      for (const row of data) {
+        if (!activeDevicesPerBuildingAndSpace[row.building_number])
+          activeDevicesPerBuildingAndSpace[row.building_number] = {};
+        activeDevicesPerBuildingAndSpace[row.building_number][row.room_id] =
+          row.device_count;
+      }
+    }
+    return activeDevicesPerBuildingAndSpace;
+  }
+
+  function subscribeToSpacesOccupancy(
+    callback: (
+      buildingNumber: number,
+      roomId: string,
+      deviceCount: number
+    ) => void
+  ) {
+    const spacesRealtimeChannel = client
+      .channel("spaces")
+      .on(
+        "postgres_changes",
+        { event: "*", schema: "public", table: "spaces_latest_states" },
+        ({ eventType, new: newData, old }) => {
+          if (eventType == "UPDATE" || eventType == "INSERT") {
+            callback(
+              newData.building_number,
+              newData.room_id,
+              newData.device_count
+            );
+          } else {
+            callback(old.building_number, old.room_id, 0);
```

Comment:
The `else` is `"DELETE"` right? Should the value then be `0` or `null`? Because basically it means we don't know if there are connected device is in a space right?
(I guess the same applies to buildings, not just spaces)

Reply:
I'm not sure we should distinguish between the two.
Currently the `activeDevices` properties in the `Building` and `Space` type are optional just to not have separate types and not do object reconstruction when we add active device data to the objects and still have TypeScript happy and have the type info correspond to what actually is in the object (it is undefined the moment we get data from the JSON files). Otherwise I supposed to have it defined as a number. In the UI `0` and `undefined` are treated uniformly.
We can get a row deletion in a table with redundant data if there are no access points for that room or building that reported their state lately. That could be a valid situation if the room has been totally closed I guess, but in this case it should probably also be handled elsewhere.